storage/repository/uri.go

```diff
@@ -10,6 +10,23 @@ import (
 	"fyne.io/fyne/v2"
 )
 
+// EqualURI returns true if the two URIs are equal.
+//
+// Since: 2.6
+func EqualURI(t1, t2 fyne.URI) bool {
+	if t1 == nil || t2 == nil {
+		return t1 == t2
+	}
+
+	u1, ok1 := t1.(*uri)
```

Comment:
URI is an interface, we cannot assert that it is always backed by our internal type.

Reply:
Indeed. I realised that I forgot to add tests for it but the reason for checking it he underlying type is that we can do a fast comparison directly without allocating the whole string. If you look at the comparison at the bottom of the function, you'll see that it doesn't compare the type asserted types but the interfaces.

Reply:
Apologies you're quite right

Reply:
No worries. I switched around the logic and added an extra comment. I hope it is clearer now :)